docs/docsite/rst/dev_guide/testing/sanity/import.rst

```diff
@@ -1,5 +1,39 @@
 import
 ======
 
-All Python imports in ``lib/ansible/modules/`` and ``lib/ansible/module_utils/`` which are not from the Python standard library
-must be imported in a try/except ImportError block.
+All Python imports in the directories specified that are not from a limited set of possibilities listed below
+must be imported in a try/except ImportError block as follows:
+
+.. code-block:: python
+
+    # Instead of 'import another_library', do:
+
+    try:
+        import another_library
+        HAS_ANOTHER_LIBRARY = True
+    except ImportError:
+        HAS_ANOTHER_LIBRARY = False
+
+
+    # Later in module code:
+
+        if not HAS_ANOTHER_LIBRARY:
+            # Needs: from ansible.module_utils.basic import missing_required_lib
+            module.fail_json(msg=missing_required_lib('another_library'))
+
+    # Later in plugin code:
+
+        if not HAS_ANOTHER_LIBRARY:
+            raise AnsibleError('another_library must be installed to use this plugin')
```

Comment:
I want to add that if the code targets a new enough Python 3 version, the example could preserve more metadata about the exception causes:
```python
    try:
        import another_library
    except ImportError as imp_exc:
        another_library_import_exc = imp_exc
    else:
        another_library_import_exc = None


    # Later in module code:

        if another_library_import_exc is not None:
            # Needs: from ansible.module_utils.basic import missing_required_lib
            module.fail_json(msg=missing_required_lib('another_library'))

    # Later in plugin code:

        if another_library_import_exc is not None:
            raise AnsibleError('another_library must be installed to use this plugin') from another_library_import_exc
```

For older Pythons the example could use `six.raise_from()` shim.

Reply:
I split up the example into modules vs. plugins, and used `traceback` to return the import traceback in `module.fail_json()`, and `six.raise_from()` for plugins (e74bec06995093c876c6fd81bc5615951641cba8).